contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc

```diff
@@ -15,30 +15,43 @@ namespace Envoy {
 namespace Extensions {
 namespace Dlb {
 
+Envoy::Network::ConnectionBalancerSharedPtr
+DlbConnectionBalanceFactory::fallback(const bool is_exact_fallback, const std::string& message) {
```

Comment:
Consider naming this to capture more of what the function is doing.

Also ISTM that we are creating lots of these load balancers, etc. in various places but might don't end up using them e.g. the return value here isn't used often

Reply:
This function aims to return software connection balance when any error occur, when returns, the balancer must be used. 

It is not load balance, it is connection balance.

Reply:
Sorry for using the incorrect term;  there are several callsites where I don't see the created fallback connection balance being returned

Reply:
Return here:

https://github.com/envoyproxy/envoy/blob/0738682f90db658be778d250ff401603ce5c87cf/source/extensions/listener_managers/listener_manager/listener_impl.cc#L778-L803

Reply:
If you look at https://github.com/envoyproxy/envoy/pull/27985/files#diff-6e97cbb906682a9017376a72808be7a265adf283870dccf8b44887698982d0f0R56-R57

We invoke fallback there and if it returns an exception we'd abort the control flow of createConnectionBalancerFromProto. But if it fallsback we'd still continue ? I think we should just return the fallback instead?

Reply:
The fallback aims to make Envoy continue work if any error about dlb parts occur. All previous error will  log as warn.

Reply:
Thanks for returning this now if there's a fallback